src/main/java/io/strimzi/kafka/bridge/tracing/OpenTelemetryHandle.java

```diff
@@ -0,0 +1,207 @@
+
+package io.strimzi.kafka.bridge.tracing;
+
+import io.jaegertracing.Configuration;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.instrumentation.kafkaclients.TracingConsumerInterceptor;
+import io.opentelemetry.instrumentation.kafkaclients.TracingProducerInterceptor;
+import io.opentelemetry.sdk.autoconfigure.OpenTelemetrySdkAutoConfiguration;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import io.strimzi.kafka.bridge.config.BridgeConfig;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.kafka.client.producer.KafkaProducerRecord;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * OpenTelemetry implementation of Tracing.
+ */
+class OpenTelemetryHandle implements TracingHandle {
+
+    static void setCommonAttributes(SpanBuilder builder, RoutingContext routingContext) {
+        builder.setAttribute("component", COMPONENT);
+        builder.setAttribute(SemanticAttributes.PEER_SERVICE, KAFKA_SERVICE);
+        builder.setAttribute(SemanticAttributes.HTTP_METHOD, routingContext.request().method().name());
+        builder.setAttribute(SemanticAttributes.HTTP_URL, routingContext.request().uri());
+    }
+
+    @Override
+    public String envName() {
+        return "OTEL_SERVICE_NAME";
+    }
+
+    @Override
+    public String serviceName(BridgeConfig config) {
+        String serviceName = System.getenv(envName());
+        if (serviceName == null) {
+            serviceName = System.getProperty("otel.service.name");
+            if (serviceName == null) {
+                // legacy purpose, use previous JAEGER_SERVICE_NAME as OTEL_SERVICE_NAME (if not explicitly set)
+                serviceName = System.getenv(Configuration.JAEGER_SERVICE_NAME);
+                if (serviceName == null) {
+                    serviceName = config.getTracingServiceName();
+                }
+                if (serviceName != null) {
+                    System.setProperty("otel.service.name", serviceName);
+                }
+            }
+        }
+        return serviceName;
+    }
+
+    @Override
+    public void initialize() {
+        OpenTelemetrySdkAutoConfiguration.initialize();
+    }
+
+    private static Tracer get() {
+        return GlobalOpenTelemetry.getTracer(COMPONENT); // TODO OK?
```

Comment:
what's this "TODO OK?" ?

Reply:
Just an early remark for me, removing it ...